darts/metrics/metrics.py

```diff
@@ -169,65 +269,457 @@ def _get_values_or_raise(
+        A deterministic or stochastic ``TimeSeries`` instance (the predictions `pred_series`).
     intersect
         A boolean for whether to only consider the time intersection between `series_a` and `series_b`
     stochastic_quantile
         Optionally, for stochastic predicted series, return either all sample values with (`stochastic_quantile=None`)
         or any deterministic quantile sample values by setting `stochastic_quantile=quantile` {>=0,<=1}.
     remove_nan_union
-        By setting `remove_non_union` to True, remove all indices from `series_a` and `series_b` which have a NaN value
-        in either of the two input series.
-    """
+        By setting `remove_non_union` to True, sets all values from `series_a` and `series_b` to `np.nan` at indices
+        where any of the two series contain a NaN value. Only effective when `is_insample=False`.
+    is_insample
+        Whether `series_a` corresponds to the `insample` series for scaled metrics.
 
-    raise_if_not(
-        series_a.width == series_b.width,
-        "The two time series must have the same number of components",
-        logger,
-    )
+    Raises
+    ------
+    ValueError
+        If `is_insample=False` and the two time series (or their intersection) do not have the same time index.
+    """
 
-    raise_if_not(isinstance(intersect, bool), "The intersect parameter must be a bool")
+    if not series_a.width == series_b.width:
+        raise_log(
+            ValueError("The two time series must have the same number of components"),
+            logger=logger,
+        )
 
-    series_a_common = series_a.slice_intersect(series_b) if intersect else series_a
-    series_b_common = series_b.slice_intersect(series_a) if intersect else series_b
+    if not isinstance(intersect, bool):
+        raise_log(ValueError("The intersect parameter must be a bool"), logger=logger)
 
-    raise_if_not(
-        series_a_common.has_same_time_as(series_b_common),
-        "The two time series (or their intersection) "
-        "must have the same time index."
-        "\nFirst series: {}\nSecond series: {}".format(
-            series_a.time_index, series_b.time_index
-        ),
-        logger,
-    )
+    make_copy = False
+    if not is_insample:
+        # get the time intersection and values of the two series (corresponds to `actual_series` and `pred_series`
+        if series_a.has_same_time_as(series_b) or not intersect:
+            vals_a_common = series_a.all_values(copy=make_copy)
+            vals_b_common = series_b.all_values(copy=make_copy)
+        else:
+            vals_a_common = series_a.slice_intersect_values(series_b, copy=make_copy)
+            vals_b_common = series_b.slice_intersect_values(series_a, copy=make_copy)
+
+        if not len(vals_a_common) == len(vals_b_common):
+            raise_log(
+                ValueError(
+                    "The two time series (or their intersection) "
+                    "must have the same time index."
```

Comment:
I would rather phrase it  (here and in the docstring) as "The two time series must have at least partially overlapping time indexes" because "same time index" is unclear?

Reply:
👍